doc/development/source/tutorial-utilities.rst

```diff
@@ -0,0 +1,146 @@
+        ...
+
+Progress bar class
+..................
+
+Class `Orange.widgets.gui.ProgressBar` is initialized with a widget and the
+number of iterations::
+
+    progress = Orange.widgets.gui.ProgressBar(self, n)
+
+is equivalent to ::
+
+     with self.progressBar(n) as progress:
+
+Direct manipulation of progress bar
+...................................
+
+The progress bar can be manipulated directly through a set of
+functions, which include:
+
+* :func:`~Orange.widgets.widget.OWWidget.progressBarInit(self)`
+* :func:`~Orange.widgets.widget.OWWidget.progressBarSet(self, f)`
+* :func:`~Orange.widgets.widget.OWWidget.progressBarFinished(self)`
+
+`progressBarInit` first method initializes the progress bar,
+`progressBarSet` sets the progress (in percents from 0 to 100),
+and `progressBarFinished` closes it.
+
+The code that uses these methods must use a try-except or try-finally block
+to make sure that the progress bar is removed if the exception is raised.
+
+Issuing warning and errors
+**************************
+
+Widgets can show information messages, warnings and errors. These are shown
+with small indicators (with tooltips) in the schema, and printed out in the
+row at the top of the widget.
+
+A widget can have multiple such messages at the same time. These are
+distinguished by different id's.
+
+A typical usage is ::
+
+    self.warning(42, "Some variables have been discretized")
+
+to issue the warning, and ::
+
+    self.warning(42)
+
+to remove it. The id 42 is arbitrary and is assigned by the programmer.
+Multiple error messages can be removed by calling with a list of id's
+(`self.warning([1, 13, 42])`).
+
+Since most widgets have only a single error message, the id can be omitted
+(in which case it defaults to 0). In this case, ::
+
+    self.warning("Some variables have been discretized")
+
+is used to issue the warning, and ::
+
+    self.warning()
+
+to remove it.
+
+Similar functions for issuing information and errors are
+`self.information(id, text)` and `self.error(id, text)`.
+
+
+Tips
+****
+
+Widgets can give the user tips about features that may not be obvious or
+exposed in the GUI.
+
+Such messages are stored in widget's class attribute `UserAdviceMessages`.
+When a widget is first shown, a message from this list is selected for display.
+If a user accepts (clicks 'Ok. Got it') the choice is recorded and the message
+is never shown again (just closing the message will not mark it as seen).
+Messages can be displayed again by pressing Shift + F1.
+
+`UserAdviceMessages` contains instances of
+:obj:`~Orange.widgets.widget.Message`. The messages contains a text and an
+id (also a string), and, optionally, an icon and an URL with further
+information.
+
+The confusion matrix widget sets up the following list::
+
+    UserAdviceMessages = [
+        widget.Message(
+                "Clicking on cells or in headers outputs the corresponding "
+                "data instances",
+                "click_cell")]
```

Comment:
This was actually correct. `"click_cell"` is the second argument, not the third. In revision, I now violated PEP-8 by additional indent in the second line to avoid confusion.